packages/block-editor/src/components/block-list/block-mobile-toolbar.js

```diff
@@ -9,11 +9,11 @@ import { ifViewportMatches } from '@wordpress/viewport';
 import BlockMover from '../block-mover';
 import VisualEditorInserter from '../inserter';
 
-function BlockMobileToolbar( { clientId } ) {
+function BlockMobileToolbar( { clientId, moverOptions } ) {
 	return (
 		<div className="editor-block-list__block-mobile-toolbar block-editor-block-list__block-mobile-toolbar">
 			<VisualEditorInserter />
-			<BlockMover clientIds={ [ clientId ] } />
+			<BlockMover clientIds={ [ clientId ] } moverOptions={ moverOptions } />
```

Comment:
A similar thing happened with the PR that introduced the ButtonAppender—the props had to be passed through several layers of components.

As more customisation of inner blocks happens this might become messy.

I don't think it's something to consider for this PR, but I do wonder if an `InnerBlocks` context might be something to consider. Particular props from `InnerBlocks` could be set on the context and picked up by deeper nested components.

Reply:
Could this be a way to achieve the original API proposed by @mtias in https://github.com/WordPress/gutenberg/issues/16637#issuecomment-516410624?

That is, a block developer would specify `<InnerBlocks horizontal>` and `BlockListBlock` would detect from `InnerBlock`/`BlockList` context whether it needs to render `<BlockMover>` inside or outside the block container.

Reply:
Yeah, that's exactly the idea. Will save us from continually expanding those in-between components when adding these kind of customisations, at the cost of a little bit more complexity.